src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,654 @@
+  for i, pattern in enumerate(ant_pattern):
+    ant_pattern_gain[i] = pattern['gain']
+
+  # Get ESC antenna information
+  esc_antenna_info = interf.EscInformation(antenna_height=esc_point['height'],
+                       antenna_azimuth=esc_point['antennaAzimuth'],
+                       antenna_gain=esc_point['antennaGain'],
+                       antenna_pattern_gain=ant_pattern_gain)
+
+  pool = Pool(processes=min(interf.NUM_OF_PROCESSES, len(protection_point)))
+
+  # ESC Category A CBSDs between 3550 - 3660 MHz within 40 km
+  protection_channels = interf.getProtectedChannels(interf.ESC_CAT_A_LOW_FREQ_HZ, 
+                          interf.ESC_CAT_A_HIGH_FREQ_HZ)
+
+  logging.debug('$$$$ Calling ESC Category A Protection $$$$')
+  # ESC algorithm
+  iapPointConstraint(protection_point, protection_channels, 
+    interf.ESC_CAT_A_LOW_FREQ_HZ, interf.ESC_CAT_A_HIGH_FREQ_HZ, 
+    grant_objects, None, esc_antenna_info, None, 
+    iap_margin_esc, interf.ProtectedEntityType.ESC_CAT_A)
+
+  # ESC Category B CBSDs between 3550 - 3680 MHz within 80 km
+  protection_channels = interf.getProtectedChannels(interf.ESC_CAT_B_LOW_FREQ_HZ, 
+                          interf.ESC_CAT_B_HIGH_FREQ_HZ)
+
+  logging.debug('$$$$ Calling ESC Category B Protection $$$$')
+  iapPointConstraint(protection_point, protection_channels, 
+    interf.ESC_CAT_B_LOW_FREQ_HZ, interf.ESC_CAT_B_HIGH_FREQ_HZ, 
+    grant_objects, None, esc_antenna_info, None, 
+    iap_margin_esc, interf.ProtectedEntityType.ESC_CAT_B)
+
+  pool.close()
+  pool.join()
+
+  return
+
+
+def performIapForGwpz(protected_entity, grant_objects, iap_margin_gwpz):
+  """Compute post IAP interference margin for GWPZ incumbent type
+
+  Routine to get protection points within GWPZ protection area and perform 
+  IAP algorithm on each protection point 
+
+  Args:
+    protected_entity: List of protection points inside protection area
+    grant_objects: a list of grant requests, each one being a dictionary 
+                   containing grant information
+    iap_margin_gwpz: Pre IAP threshold value for GWPZ incumbent type(mW/RBW)
+  """
+
+  logging.debug('$$$$ Getting GRID points for GWPZ Protection Area $$$$')
+  # Get Fine Grid Points for a GWPZ protection area
+  protection_points = utils.GridPolygon(protected_entity)
+
+  gwpz_freq_range = protected_entity['deploymentParam']\
+                      ['operationParam']['operationFrequencyRange']
+  gwpz_low_freq = gwpz_freq_range['lowFrequency']
+  gwpz_high_freq = gwpz_freq_range['highFrequency']
+
+  gwpz_region = protected_entity['zone']['features'][0]\
+                                 ['properties']['clutter']
+
+  # Get channels over which area incumbent needs partial/full protection
+  protection_channels = interf.getProtectedChannels(gwpz_low_freq, gwpz_high_freq)
+
+  # Apply IAP for each protection constraint with a pool of parallel processes.
+  pool = Pool(processes=min(interf.NUM_OF_PROCESSES, len(protection_points)))
```

Comment:
same comment about pool here and following functions

Reply:
Updated   to pool.map construct